cloud-controller-manager/do/loadbalancers.go

```diff
@@ -388,16 +388,26 @@ func buildHealthCheck(service *v1.Service) (*godo.HealthCheck, error) {
 	if err != nil {
 		return nil, err
 	}
+	port := service.Spec.Ports[0]
 
 	// if no health check protocol is specified, use the protocol used for
-	// load balancer traffic.
+	// load balancer traffic, unless TLS passthrough is enabled, then default
+	// to the protocol of the health check port.
 	if healthCheckProtocol == "" {
 		protocol, err := getProtocol(service)
 		if err != nil {
 			return nil, err
 		}
 
-		healthCheckProtocol = protocol
+		tlsPassThrough := getTLSPassThrough(service)
+
+		if tlsPassThrough || (protocol != "https") {
+			healthCheckProtocol = protocol
+		} else {
+			// We already validate that the port.Protocol is TCP, so we
+			// can use it as-is here.
+			healthCheckProtocol = strings.ToLower(string(port.Protocol))
```

Comment:
I think this should simply assign `protocolTCP` as that constant is specifically designed to match the values that the DO API expects.